warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java

```diff
@@ -692,4 +691,8 @@ private boolean isInfinite() {
     public void setNode(JexlNode currNode) {
         myNode = currNode;
     }
+    
+    private String nodeToKey(JexlNode node) {
+        return JexlStringBuildingVisitor.buildQueryWithoutParse(TreeFlatteningRebuildingVisitor.flatten(node), true);
```

Comment:
we call this a lot.  Is it possible to flatten the tree once?

Reply:
Per our conversation, I verified that in all of our test cases the node passed in to nodeToKey is already flattened.  We should be able to safely remove the flatten call from this method.